### PR TITLE
feat: switch the service about the elections entity using MediatR Pat…

### PR DESCRIPTION
**Pull request by VeneraGoxhufi**
…tern

### CODE REVIEW

**Review by ArianitSGashi — APPROVED** [review]

File: src/OnlineVotingS.Application/Services/Election/Handlers/Commands/CreateElectionsHandler.cs [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+    }
+
+    public async Task<Elections> Handle(CreateElectionsCommand request, CancellationToken cancellationToken)
+    {
+        try
+        {
+            var elections = _mapper.Map<Elections>(request.ElectionDto);
+            await _electionsRepository.AddAsync(elections);
+
+            _logger.LogInformation("Elections created successfully with ID {ElectionsId}.", elections.ElectionID);
```
Comment (AdhurimUP) [inline src/OnlineVotingS.Application/Services/Election/Handlers/Commands/CreateElectionsHandler.cs]: no need for this information

File: src/OnlineVotingS.Application/Services/Election/Handlers/Commands/DeleteElectionsHandler.cs [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+    }
+
+    public async Task<bool> Handle(DeleteElectionsCommand request, CancellationToken cancellationToken)
+    {
+        try
+        {
+            var exists = await _electionsRepository.ExistsAsync(request.ElectionId);
+            if (!exists)
+            {
+                _logger.LogWarning("Elections with ID {ElectionId} not found.", request.ElectionId);
```
Comment (AdhurimUP) [inline src/OnlineVotingS.Application/Services/Election/Handlers/Commands/DeleteElectionsHandler.cs]: no need for this logger

File: src/OnlineVotingS.Application/Services/Election/Handlers/Commands/DeleteElectionsHandler.cs [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+                throw new KeyNotFoundException($"Elections with ID {request.ElectionId} not found.");
+            }
+
+            await _electionsRepository.DeleteAsync(request.ElectionId);
+
+            _logger.LogInformation("Elections with ID {ElectionId} deleted successfully.", request.ElectionId);
```
Comment (AdhurimUP) [inline src/OnlineVotingS.Application/Services/Election/Handlers/Commands/DeleteElectionsHandler.cs]: no need for this logger information

File: src/OnlineVotingS.Application/Services/Election/Handlers/Commands/UpdateElectionsHandler.cs [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+    }
+
+    public async Task<Elections> Handle(UpdateElectionsCommand request, CancellationToken cancellationToken)
+    {
+        try
+        {
+            var elections = await _electionsRepository.GetByIdAsync(request.ElectionDto.ElectionID);
+            if (elections == null)
+            {
+                _logger.LogWarning("Elections with ID {ElectionId} not found.", request.ElectionDto.ElectionID);
```
Comment (AdhurimUP) [inline src/OnlineVotingS.Application/Services/Election/Handlers/Commands/UpdateElectionsHandler.cs]: no need for this

File: src/OnlineVotingS.Application/Services/Election/Handlers/Commands/UpdateElectionsHandler.cs [hunk 5]
```diff
@@ -0,0 +1,46 @@
...
+                throw new KeyNotFoundException($"Elections with ID {request.ElectionDto.ElectionID} not found.");
+            }
+
+            _mapper.Map(request.ElectionDto, elections);
+            await _electionsRepository.UpdateAsync(elections);
+
+            _logger.LogInformation("Elections with ID {ElectionId} updated successfully.", request.ElectionDto.ElectionID);
```
Comment (AdhurimUP) [inline src/OnlineVotingS.Application/Services/Election/Handlers/Commands/UpdateElectionsHandler.cs]: no need for this

File: src/OnlineVotingS.Application/Services/Election/Handlers/Queries/GetElectionByIdHandler.cs [hunk 6]
```diff
@@ -0,0 +1,40 @@
...
+    }
+
+    public async Task<Elections> Handle(GetElectionsByIdQuery request, CancellationToken cancellationToken)
+    {
+        try
+        {
+            var campaign = await _electionRepository.GetByIdAsync(request.ElectionID);
+            if (campaign == null)
+            {
+                _logger.LogWarning("Election with ID {ElectonId} not found.", request.ElectionID);
```
Comment (AdhurimUP) [inline src/OnlineVotingS.Application/Services/Election/Handlers/Queries/GetElectionByIdHandler.cs]: no need for this

File: src/OnlineVotingS.Application/Services/Election/Requests/Commands/CreateElectionsCommand.cs [hunk 7]
```diff
@@ -0,0 +1,15 @@
...
+
+namespace OnlineVotingS.Application.Services.Election.Requests.Commands;
+
+public class CreateElectionsCommand : IRequest<Elections>
+    {
+        public ElectionsPostDTO ElectionDto { get; }
+
+        public CreateElectionsCommand(ElectionsPostDTO electionDto)
+        {
+        ElectionDto = electionDto;
```
Comment (AdhurimUP) [inline src/OnlineVotingS.Application/Services/Election/Requests/Commands/CreateElectionsCommand.cs:13]: align code please

File: src/OnlineVotingS.Application/Services/Election/Requests/Commands/DeleteElectionsCommand.cs [hunk 8]
```diff
@@ -0,0 +1,18 @@
...
+
+namespace OnlineVotingS.Application.Services.Election.Requests.Commands;
+
+public class DeleteElectionsCommand : IRequest<bool>
+    {
+        public int ElectionId { get; }
+
+        public DeleteElectionsCommand(int electionId)
+        {
+        ElectionId = electionId;
```
Comment (AdhurimUP) [inline src/OnlineVotingS.Application/Services/Election/Requests/Commands/DeleteElectionsCommand.cs:16]: align code please beautify it

**Pull request merged**